### ISSUE TITLE
Bootify CAS application context

### ISSUE THREAD

**Issue by mmoayyed** [Jasig/cas#1577]
Take advantage of Spring Boot, to configure the Spring cloud server/client config, specifically. Boot offers a hassle-free xml-free configuration style of the webapp context is able to autoconfigure most of the CAS internals via its autoconfiguration annotations. It also allows CAS to turn into an executable war via an embedded servlet container.

**Comment by mmoayyed** [comment]
Boot config functional:
https://github.com/Unicon/cas/tree/bootiful

Few caveats:
1. Way too much magic :) 
2. Running the web application in an embedded container like jetty via gretty somehow reloads the context twice. Not sure why yet.
3. Starting to weed out custom initializers.

**Comment by dima767** [comment]
Boot already includes embedded servlet container by default. Why do you
need to run it via gretty?

wrt "magic" - embrace it. If we didn't embrace "magic" long time ago, we'd
still be programming in assembly language :-)

**Comment by mmoayyed** [comment]
I was trying to figure out if it works in all 3 modes. Looks like it fails to load with gretty, or it somehow loads the embedded jetty context twice for some reason. Havent tried the built-in tomcat container.

**Comment by dima767** [comment]
I don't know, but this might be of help, if anything:
http://akhikhl.github.io/gretty-doc/spring-boot-support.html

**Comment by mmoayyed** [comment]
Thanks. That does help. 

Got everything working. Working quirks, internals and webflow config.

**Comment by mmoayyed** [comment]
Had to do a few tricks to make sure when CAS runs in an embedded container, and one that does not explode the war (like the executable war), all @Value annotations that point to classpath resources work correctly.

**Comment by mmoayyed** [comment]
Got about 6-7 major items to work out before I can produce anything useful. also, lots of tests and I am updating docs as I work through these. I think this sort of change might be appropriate for 4.4, but if we can fit into 4.3, all the better.

**Comment by mmoayyed** [comment]
Boot config fairly functional. What I will do, once I get a bit closer, is I will move the changes to a branch here in this repo so that others can easily pull down and review, etc. Same process we are doing with MFA. Presently on the TODO list:

1. Consolidate/Remove CAS initializers and have boot handle the context init tasks
2. Test all modules, sort out classpath issues with resources with running in embedded mode. 
3. Remove/consolidate cas-servlet and dispatcher config. Let boot handle all of that. 
4. Consolidate messaging config for UI. Let boot handle all of that. 
5. Test all modules.
6. Test the overlay as we have it today with boot. 
7. Update docs
8. Start with server/client cloud config

**Comment by mmoayyed** [comment]
- [X] Consolidate/Remove CAS initializers and have boot handle the context init tasks
- [X] Test all modules, sort out classpath issues with resources with running in embedded mode.
- [X] Remove/consolidate cas-servlet and dispatcher config. Let boot handle all of that.
- [X] Consolidate messaging config for UI. Let boot handle all of that.
- [X] Test all modules.
- [ ] Test the overlay as we have it today with boot.
- [X] Update docs
- [ ] Start with server/client cloud config

**Comment by mmoayyed** [comment]
Spring cloud perfectly functional. Working on docs. Probably should do another code review session soon to go over changes. It's way too cool.

**Comment by mmoayyed** [comment]
Moving the milestone back to 4.3, since I think we can make it.

**Comment by mmoayyed** [comment]
Status update: everything is fully functional. Need to figure out the cloud monitor thingy with AMQP and test the standard overlay. We should ready for a code review session soon.

**Comment by mmoayyed** [comment]
This is completely ready. cloud-monitor changes with AMQP arent ready and released yet by boot. They are only in milestone mode, and not compatible with the other GA releases of boot. We can do without for now, but will have to keep an eye out towards its release so we can update. The changes to enable the monitor are extremely minimal.

I am going to start with docs, and then will push changes to a branch here. I would want us to merge MFA first before starting with boot, as maintaining two giant branches in parallel would be very painful.

**Comment by dima767** [comment]
Has the regular deployer's maven war overlay with "bootiful" artifacts as
dependencies been tested?

On Tue, Mar 15, 2016 at 04:42, Misagh Moayyed <notifications@github.com> wrote:
This is completely ready. cloud-monitor changes with AMQP arent ready and
released yet by boot. They are only in milestone mode, and not compatible with
the other GA releases of boot. We can do without for now, but will have to keep
an eye out towards its release so we can update. The changes to enable the
monitor are extremely minimal.

I am going to start with docs, and then will push changes to a branch here. I
would want us to merge MFA first before starting with boot, as maintaining two
giant branches in parallel would be very painful.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub:
https://github.com/Jasig/cas/issues/1577#issuecomment-196721783
[https://github.com/Jasig/cas/issues/1577#issuecomment-196721783][https://github.com/notifications/beacon/AAAxVUplG2aylJkSLw94W4qKFsv5F9eUks5ptnEIgaJpZM4Hgt8l.gif]

**Comment by mmoayyed** [comment]
On the todo list, but that might be a separate task from here. I'd forgotten that the mgmt app should also support thymeleaf. Almost done there, then the docs, then the overlay.

**Comment by dima767** [comment]
Roger that. We need to keep in mind that the deployer's overlays must work as is
today (without being even aware that they depend on the "bootiful" CAS
artifacts) with the usual property placeholder configurer, etc. - until at least
we come up perhaps with the different deployment configuration model

On Tue, Mar 15, 2016 at 07:09, Misagh Moayyed <notifications@github.com> wrote:
On the todo list, but that might be a separate task from here. I'd forgotten
that the mgmt app should also support thymeleaf. Almost done there, then the
docs, then the overlay.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub:
https://github.com/Jasig/cas/issues/1577#issuecomment-196767798
[https://github.com/Jasig/cas/issues/1577#issuecomment-196767798][https://github.com/notifications/beacon/AAAxVRlJ9aEzqI_lFN9tOd79mxceWs-pks5ptpNngaJpZM4Hgt8l.gif]

**Comment by mmoayyed** [comment]
Management app done, and docs are updated. Working on UX changes as it seems like I have broken a few things with the move towards thymeleaf :)
